src/pywrapper.cpp

```diff
@@ -102,26 +116,61 @@ bool PyWrapper::init()
     // communication channel for I/O Intr value exchange
     PyImport_AppendInittab("pydev", &PyInit_pydev);
 
+    // Py_SetProgramName(L"python3.7");
+    std::cerr << "Initalising python3" << std::endl;
     Py_Initialize();
+    std::cerr << "Initialsing threads" << std::endl;
     PyEval_InitThreads();
 
+    std::cerr << "Initialsing dics" << std::endl;
+
+#if PY_MAJOR_VERSION == 3
+
+    /*
+     * IN Python 3 the
+     */
+    auto m = PyImport_AddModule("__main__");
+    assert(m);
+    globDict = PyModule_GetDict(m);
+    assert(globDict);
```

Comment:
Let's move assert(globDict) and assert(locDict) *after* the endif to capture both Python2 and Python3

Reply:
Agreed